plugins/modules/perlmod_install_info.py

```diff
@@ -0,0 +1,383 @@
+            dependencies |= set(m for m in match.group(1).split(', ')
+                                if m not in modules)
+    return (found, dependencies)
+
+
+def find_modules(amodule, result, errors, names=None, update=None):
+    """Stores results in `result`, returns set of missing modules."""
+    params = amodule.params
+    names = set(params['name'] if names is None else names)
+    update = params['update'] if update is None else update
+
+    try_installed = params['try_installed']
+    try_dnf = amodule.get_bin_path('dnf') is not None \
+        if params['try_dnf'] == 'auto' else boolean(params['try_dnf'])
+    if not try_dnf:
+        try_yum = amodule.get_bin_path('yum') is not None \
+            if params['try_yum'] == 'auto' else boolean(params['try_yum'])
+    try_apt = amodule.get_bin_path('apt-file') is not None \
+        if params['try_apt'] == 'auto' else boolean(params['try_apt'])
+    try_cpanm = amodule.get_bin_path('cpanm') is not None \
+        if params['try_cpanm'] == 'auto' else boolean(params['try_cpanm'])
+
+    found = set()
+
+    if names and try_installed:
+        this_found = set(n for n in names if check_installed(amodule, n))
+        result['installed'] = result.get('installed', set()) | this_found
+        found |= this_found
+        names -= this_found
+
+    if names and (try_dnf or try_yum):
+        which_cmd = 'dnf' if try_dnf else 'yum'
+        (this_found, packages) = dnf_or_yum(amodule, which_cmd, update, names)
+        result[which_cmd] = result.get(which_cmd, set()) | packages
+        found |= this_found
+        names -= this_found
+
+    if names and try_apt:
+        (this_found, packages) = apt(amodule, update, names)
+        result['apt'] = result.get('apt', set()) | packages
+        found |= this_found
+        names -= this_found
+
+    if names and try_cpanm:
+        (this_found, dependencies) = cpanm(amodule, names)
+        result['cpanm'] = result.get('cpanm', set()) | this_found
+        found |= this_found
+        names -= this_found
+        dependencies -= found
+        names |= find_modules(amodule, result, errors, names=dependencies,
+                              update=False)
+
+    return names
+
+
+def run_module():
+    """Does all the work."""
+    module_args = dict(
+        name=dict(type='list', elements='str', required=True),
+        try_installed=dict(type='bool', required=False, default=True),
+        try_dnf=dict(type='str', required=False, default='auto',
+                     choices=['auto', 'true', 'false']),
+        try_yum=dict(type='str', required=False, default='auto',
+                     choices=['auto', 'true', 'false']),
+        try_apt=dict(type='str', required=False, default='auto',
+                     choices=['auto', 'true', 'false']),
+        try_cpanm=dict(type='str', required=False, default='auto',
+                       choices=['auto', 'true', 'false']),
+        update=dict(type='bool', required=False, default=False),
```

Comment:
It is common practice to suppress `required=True` from the the arg spec, since it is redundant.
```suggestion
        try_installed=dict(type='bool', default=True),
        try_dnf=dict(type='str', default='auto',
                     choices=['auto', 'true', 'false']),
        try_yum=dict(type='str', default='auto',
                     choices=['auto', 'true', 'false']),
        try_apt=dict(type='str', default='auto',
                     choices=['auto', 'true', 'false']),
        try_cpanm=dict(type='str', default='auto',
                       choices=['auto', 'true', 'false']),
        update=dict(type='bool', default=False),
```

Reply:
It's not a "common practice" I agree with for two reasons. First, specifying the values in a consistent format makes it easier for the eye to scan them, and second, I don't see any reason to make people work harder to remember what the default for the `required` parameter is when reading the code.